src/Symfony/Bundle/FrameworkBundle/Resources/config/web.xml

```diff
@@ -17,6 +17,29 @@
             <argument type="service" id="logger" on-invalid="ignore" />
         </service>
 
+        <service id="argument_resolver" class="Symfony\Component\HttpKernel\Controller\ArgumentResolver" public="false">
+            <argument type="service" id="argument_metadata_factory" />
+            <argument type="collection" />
+        </service>
+
+        <service id="argument_metadata_factory" class="Symfony\Component\HttpKernel\ControllerMetadata\Argument\ArgumentMetadataFactory" public="false" />
+
+        <service id="argument_value_resolver.argument_from_attribute" class="Symfony\Component\HttpKernel\Controller\ArgumentValueResolver\ArgumentFromAttribute" public="false">
+            <tag name="controller_argument.value_resolver" priority="100" />
+        </service>
+
+        <service id="argument_value_resolver.argument_is_request" class="Symfony\Component\HttpKernel\Controller\ArgumentValueResolver\ArgumentIsRequest" public="false">
+            <tag name="controller_argument.value_resolver" priority="50" />
+        </service>
+
+        <service id="argument_value_resolver.default_argument_value" class="Symfony\Component\HttpKernel\Controller\ArgumentValueResolver\DefaultArgumentValue" public="false">
+            <tag name="controller_argument.value_resolver" priority="-100" />
+        </service>
+
+        <service id="argument_value_resolver.variadic_argument_from_attribute" class="Symfony\Component\HttpKernel\Controller\ArgumentValueResolver\VariadicArgumentFromAttribute" public="false">
+            <tag name="controller_argument.value_resolver" priority="-150" />
+        </service>
```

Comment:
What about merging the last two as being the "fallback" behavior based on reflection? That could even be "hardcoded" (not sure).